libweston/backend-rdp/rdprail.c

```diff
@@ -1001,6 +1001,14 @@ rail_client_LanguageImeInfo_callback(bool freeOnly, void *arg)
 						__func__, new_keyboard_layout,
 						settings->KeyboardType,
 						settings->KeyboardSubType);
+
+                                rdp_debug_error(b, "%s: Resetting default keymap\n", __func__);
+                                keymap = xkb_keymap_new_from_names(peer_ctx->item.seat->compositor->xkb_context,
+                                                                   &peer_ctx->item.seat->compositor->xkb_names,
+                                                                   0);
+                                weston_seat_update_keymap(peer_ctx->item.seat, keymap);
+                                xkb_keymap_unref(keymap);
+                                settings->KeyboardLayout = new_keyboard_layout;

```

Comment:
Ideally "new_keyboard_layout" should reflect the default.

Reply:
Unfortunately that's not always possible because you may have a xkb layout in weston.ini that doesn't have a windows counter-part. E.g. us-intl with altgr dead keys is a pretty popular one for europeans using us keyboards that doesn't exist in windows unless you use MSKLC layouts.

Still you need to save the layout in `settings->KeyboardLayout` otherwise the `if (new_keyboard_layout != settings->KeyboardLayout)` a few lines above will fail and prevent you to actually change layout.